### ISSUE TITLE
Just looking for clarification on usage

### ISSUE THREAD

**Issue by seandfeeney** [jswanner/DontFuckWithPaste#112]
Is it my understanding that you have to manually add each site to the "blacklist" in order for this to work? I think I am confused about the terminology of "blacklist" in this context.

Also, I recommend considering removing the language blacklist or whitelist because they unintentionally relay an inappropriate racial undertone.  Some alternatives to consider that I feel are easier for the layperson to understand regardless of your stance on the matter.

Exclude List
Allow List

Block List
Allow List

Reject List
Permit List

Minus List
Plus List

Exclusions
Inclusions

**Comment by jswanner** [comment]
@seandfeeney,

> Also, I recommend considering removing the language blacklist or whitelist because they unintentionally relay an inappropriate racial undertone.

Fair point, the language of the README has been changed.

> Is it my understanding that you have to manually add each site to the "blacklist" in order for this to work? I think I am confused about the terminology of "blacklist" in this context.

**TL;DR:**

By default the extension doesn't do anything. When you come across a site that is blocking copy/cut/paste, click on the extension's button (optionally adjust the pattern) then click the "Save" button, at that point copy/cut/paste functionality _should_ be working.

**Longer version**

So there aren't really any "blacklists" and "whitelists," the extension is either active or inactive for the URL you are visiting, by default it is inactive. There also isn't a list of sites where the extension is active and another list where the extension is inactive. Instead, there's only a list of patterns and the extension is active if the current URL matches any of those patterns, by default there are no patterns which is why the extension is inactive by default.

When the extension is active then it intercepts copy/cut/paste events before the page does and forces the browser's default behavior for those events. You activate the extension by adding patterns to the extension's settings that match the URL of the pages you want it to be active. For instance, if you add a pattern of `.*` then the extension will always be active but you will have a poor browsing experience when you are on sites that do useful things with those events (like GitHub turning a file paste into a file upload).

When the extension is inactive (as in there are no patterns that match the current URL) and you click on the extension's icon button, it will open a popup where you can add patterns and it will be populated with one that matches the domain of the current URL. Generally speaking, the suggested pattern will be sufficient and so all you need to do is click on the "Save" button in the popup; at that point the extension will be active and it will reactivate any time you visit a URL with that domain.

When the extension is active and you click on the extension's icon button, the popup will be populated with all patterns that match the current URL. If you remove all the matching patterns and click save, then that will deactivate the extension.

**Caveat**

If a site is also blocking the keyboard shortcuts associated with copy, cut or paste, this extension does not circumvent that, you will need to right click or use the browser's Edit menu to accomplish those tasks.

**Comment by acino1answers** [comment]
Anyone please do answer me this!!
Will the site owner or someone know??
If I paste text ...in the site where paste is blocked???
And 1 more doubt if i on extension and paste later on I off extension than someone shall know this also??

**Comment by jswanner** [comment]
@acino1answers I'm sorry, I'm not quite able to understand the problem you are experiencing. 

Can you open a new issue, specifying the site you are having trouble with and any other details about what you are seeing?

**Comment by acino1answers** [comment]
Hello sir!!
I have no issues with app!!
I just have one small question!!
Please do answer this!!
If I use the extension and paste the text .
Will the site owner know that I pasted the text ???...


Example
-- there is some site name "A"
So site  "A" had blocked the option paste ,so now I used the extension and
pasted the text....
So will the site owner aware of my actions??

Thing is I got online exam!!
And I am not that fast on keyboard..
And the site where I write exam don't allow me to paste!!!
So if I paste using the extension will anyone know????

**Comment by jswanner** [comment]
> If I use the extension and paste the text . Will the site owner know that I pasted the text ???...

Most likely not. If you used the keyboard shortcut for initiating the "paste" event (ctrl-v/cmd-v), they could have event listeners registered for those and record that activity (I've never noticed a site that does that, but it's theoretically possible). If you trigger "paste" from the Edit menu or from context menu (right click) then no, if the site had event listeners for "paste" events this extension prevents the site from receiving those events.

**Comment by acino1answers** [comment]
I checked my exam site there is no such events...when I inspected elements!!
https://ks.wjx.top/m/30040783.aspx
That's my sample test ..
If you have a minute could u please confirm it !!!
It's been so hard since lockdown even I know the answers I can't type
within the time limit!!!
Please do have a look at my exam site
Thank you so much!!!
